nemo/collections/asr/parts/submodules/multi_head_attention.py

```diff
@@ -362,3 +609,52 @@ def forward(self, x, cache_len=0):
         if self.dropout_emb:
             pos_emb = self.dropout_emb(pos_emb)
         return self.dropout(x), pos_emb
+
+
+class ChunkedRelPositionalEncoding(PositionalEncoding):
+    """Relative positional encoding for sliding window attention or chunked attention.
+    See above for relative positional encoding based on Transformer-XL paper
+    Args:
+        left_chunk_size (int): number of frames to in past chunks
+        chunk size (int): number of frames (max frames if using multimode) in current chunk
+        d_model (int): embedding dim
+        dropout_rate (float): dropout rate
+        max_len (int): maximum input length
+        xscale (bool): whether to scale the input by sqrt(d_model)
+        dropout_rate_emb (float): dropout rate for the positional embeddings
+    """
+
+    def __init__(self, att_context_size, **kwargs):
+        super(ChunkedRelPositionalEncoding, self).__init__(**kwargs)
+        self.left_context = att_context_size[0]
+        self.right_context = att_context_size[1]
+
+    def extend_pe(self, length, device):
+        """Reset and extend the positional encodings only at the beginning"""
+        if hasattr(self, 'pe'):
+            return
+
+        positions = torch.arange(
+            self.left_context, -self.right_context - 1, -1, dtype=torch.float32, device=device
+        ).unsqueeze(1)
+        self.create_pe(positions=positions)
+        self.center_pos = torch.tensor(self.left_context, dtype=torch.int32, device=device)
```

Comment:
I believe Boris updated it so this is no longer cached, please use the current Nemo code to see how to compute it on the fly so if can be traced properly

Reply:
For this class this would only be updated once anyway (as length doesn't matter), so I don't think there's a problem